### ISSUE TITLE
Could not load file or assembly 'System.Numerics.Vectors, Version=4.1.3.0, or one of its dependencies.

### ISSUE THREAD

**Issue by Woudjee** [dotnet/runtime#96980]
I have exactly the problem described in this post: https://github.com/dotnet/runtime/issues/26370 and since it is unfortunately closed, I cannot reply anymore, hence this post. I have followed the steps provided, but my application still does not run.

Concretely:

- I have a .NET Framework 4.8 Web API
- This project references a different project that contains logic (let's call it the ExportEngine project)
- The ExportEngine project references System.Numerics.Vectors
- When trying to run the application, I get the following error after implementing https://stackoverflow.com/questions/1091853/error-message-unable-to-load-one-or-more-of-the-requested-types-retrieve-the-l: "Could not load file or assembly 'System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)"
- In the .csproj file of my web api, I have specified:

```
<PropertyGroup>
    <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
</PropertyGroup>
```

And I have updated the `System.Numerics.Vectors` version to 4.1.3.0 instead of 4.1.4.0, which now looks like:

```
<Reference Include="System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
      <HintPath>..\packages\System.Numerics.Vectors.4.5.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
</Reference>
```

@eerhardt I saw you had some great insights on the previous posts, as well as @saucecontrol. Any suggestions as to what I am missing?

**Comment by huoyaoyuan** [comment]
> And I have updated the `System.Numerics.Vectors` version to 4.1.3.0 instead of 4.1.4.0, which now looks like:

This should be incorrect. The assembly file in the nuget package has version number of 4.1.4.0 in it's information.

The original issue (https://github.com/dotnet/runtime/issues/26370#issuecomment-395185748) suggest that you should add a direct `PackageReference` element to `System.Numerics.Vectors` in your entry project.

**Comment by Woudjee** [comment]
Aha, perhaps I missed that! How exactly do I do that?

**Comment by Woudjee** [comment]
Changing the `Output type` to a `Class Library` resolves the build error, but obviously afterwards the application does not start anymore because the "RunCommand property is not defined". I suppose I should not change this. But I'm still unsure what to change. No one online seems to have this issue (everyone online seems to be already using .NET Core). Unfortunately I am tied to . NET Framework in the context of this API.

**Comment by huoyaoyuan** [comment]
I have no more knowledge about ASP.NET on .NET Framework. You can consider to ask at https://github.com/dotnet/aspnetcore/discussions.

**Comment by lextm** [comment]
@huoyaoyuan ASP.NET 4.x projects should stay with the old project format, so don't suggest anyone to switch to SDK style projects or they will be out of support scope.

**Comment by lextm** [comment]
From [this comment](https://github.com/dotnet/runtime/issues/26370#issuecomment-395185748) @eerhardt has made it very clear what are the options for anyone hitting this issue.

Since you are still working on the old ASP.NET projects, you must use option 1 ("Reference the dependency (System.Numerics.Vectors) directly in the Desktop project"). This forces MSBuild to copy the right assembly to the right place and the error should disappear.

**Comment by huoyaoyuan** [comment]
> ASP.NET 4.x projects should stay with the old project format

Why? The old project format should be avoided as possible, unless there's unsupported functionality from new SDK.

**Comment by lextm** [comment]
> Why? The old project format should be avoided as possible, unless there's unsupported functionality from new SDK.

Well, it happens to be the impossible and unsupported part, https://stackoverflow.com/a/69750244/11182

**Comment by jeffhandley** [comment]
@Woudjee Did you overcome this by following the guidance to "Reference the dependency (System.Numerics.Vectors) directly in the Desktop project"? I understand we might have sent you down another path with the SDK projects, but from what I can tell that original guidance was still applicable.

**Comment by Woudjee** [comment]
It has been a while since I worked on this problem. If I remember correctly, the upgrading to SDK-style project resolved this and many other similar issues. In contrary to what I just read, I really like the upgrade assistant for older framework projects.

Regarding the web application. I did not get this working and ended up coding my own. That one is working great now. 👍 it is essentially a http server that works the same as a .NET Core web application. (But most likely less secure, which is fine in my use case).

And I do remember something about indeed referencing certain projects directly to resolve issues. But that was for I believe XBim (different package). The problem there was something in their configuration that prevented certain dependencies to be copied to the correct bin folder. Perhaps similar to this as well.

Op 19 jul 2024 om 13:09 heeft Jeff Handley ***@***.***> het volgende geschreven:

﻿

@Woudjee<https://github.com/Woudjee> Did you overcome this by following the guidance to "Reference the dependency (System.Numerics.Vectors) directly in the Desktop project"? I understand we might have sent you down another path with the SDK projects, but from what I can tell that original guidance was still applicable.
